com.unity.netcode.gameobjects/Runtime/Connection/NetworkConnectionManager.cs

```diff
@@ -567,14 +566,29 @@ private void SendConnectionRequest()
                 if (MessageManager.MessageTypes[index] != null)
                 {
                     var type = MessageManager.MessageTypes[index];
-                    message.MessageVersions[index] = new MessageVersionData
+                    var messageVersionData = new MessageVersionData
```

Comment:
This whole group of logic feels like it maybe would make sense on the data structure?

Something like two constructors, one that is for `DistributedAuthorityMode == false` and takes the hash and the version, and one that is for `DistributedAuthorityMode == true` that takes the type and the version, and then calculates the hash internally. That way if the type is important, the hash can never be wrong. That would also mean the logic of do we or do we not send the type is better encapsulated and consistent.

Reply:
Hmmm... let me fiddle with that idea...seems a bit cleaner indeed.